code/__HELPERS/mobs.dm

```diff
@@ -282,10 +282,24 @@ GLOBAL_LIST_EMPTY(species_list)
 		checked_health["health"] = health
 	return ..()
 
-/proc/do_after(mob/user, delay, atom/target = null, needhand = TRUE, progress = TRUE, datum/callback/extra_checks = null, stayStill = TRUE)
+/proc/do_after(mob/user, delay, atom/target = null, needhand = TRUE, progress = TRUE, datum/callback/extra_checks = null, stayStill = TRUE, required_skill = null, required_skill_level = SKILLLEVEL_UNSKILLED, list/skill_delay_mult_scaling = list())
 	if(!user)
 		return FALSE
 	var/atom/target_loc = null
+	if(!isnull(required_skill) && usesSkills(user))
+		var/datum/skill/user_skill = find_skill(user, required_skill)
+		if(skill_check(user, required_skill, required_skill_level))
+			if(skill_delay_mult_scaling.len > 0)
```

Comment:
```suggestion
			if(skill_delay_mult_scaling.len)
```

Reply:
> if you want to i can commit these directly @nmajask

Make a PR with the changes so I can just merge that @tattax